dotCMS/src/main/java/com/dotcms/rest/api/v1/storage/ContentStorageResource.java

```diff
@@ -0,0 +1,244 @@
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+/**
+ * Resource to expose the storage and metadata
+ * @author jsanca
+ */
+@Path("/v1/storage")
+public class ContentStorageResource {
+
+    private final WebResource    webResource;
+    private final FileStorageAPI fileStorageAPI;
+    private final MultiPartUtils multiPartUtils;
+    private final ContentStorageHelper  contentStorageHelper;
+    private final ContentletMetadataAPI contentletMetadataAPI;
+
+    @VisibleForTesting
+    protected ContentStorageResource(final FileStorageAPI fileStorageAPI,
+                                     final WebResource webResource,
+                                     final MultiPartUtils multiPartUtils,
+                                     final ContentStorageHelper  contentStorageHelper,
+                                     final ContentletMetadataAPI contentletMetadataAPI) {
+
+        this.fileStorageAPI = fileStorageAPI;
+        this.webResource    = webResource;
+        this.multiPartUtils = multiPartUtils;
+        this.contentStorageHelper  = contentStorageHelper;
+        this.contentletMetadataAPI = contentletMetadataAPI;
+    }
+
+    public ContentStorageResource() {
+        this(APILocator.getFileStorageAPI(), new WebResource(),
+                new MultiPartUtils(), new ContentStorageHelper(), APILocator.getContentletMetadataAPI());
+    }
+
+    @GET
+    @JSONP
+    @NoCache
+    @Produces(MediaType.APPLICATION_OCTET_STREAM)
+    public final Response getFileFromStorage(@Context final HttpServletRequest request,
+                                           @Context final HttpServletResponse response,
+                                           @QueryParam("path") final String path) {
+
+
+        final String storageType = Config.getStringProperty("DEFAULT_STORAGE_TYPE", StorageType.FILE_SYSTEM.name());
+
+        Logger.debug(this, ()-> "Getting the file: " + path + " from the storage: " + storageType);
+
+        final File   file        = this.fileStorageAPI.getStorageProvider().getStorage(storageType).pullFile("files", path);
+        final String fileName    = FilenameUtils.getName(path);
+
+        return Response.ok(
+                new StreamingOutput() {
+                    @Override
+                    public void write(OutputStream output)  {
+                        try (ZipOutputStream zoutput = new ZipOutputStream(output);
+                                InputStream fileInput = Files.newInputStream(file.toPath())) {
+
+                            int   bytesRead      = 0;
+                            final byte[] buffer  = new byte[512];
+                            zoutput.putNextEntry(new ZipEntry(path));
+
+                            while ((bytesRead = fileInput.read(buffer)) > 0) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid assignments in operands](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5547873)